src/github/tests/test_utils.rs

```diff
@@ -0,0 +1,193 @@
+use std::collections::{HashMap, HashSet};
+use std::rc::Rc;
+
+use derive_builder::Builder;
+use rust_team_data::v1::{GitHubTeam, Person, TeamGitHub, TeamKind};
+
+use crate::github::api::{
+    BranchProtection, GithubRead, Repo, RepoTeam, RepoUser, Team, TeamMember, TeamPrivacy,
+};
+use crate::github::{api, SyncGitHub, TeamDiff};
+
+const DEFAULT_ORG: &str = "rust-lang";
+
+/// Represents the contents of rust_team_data state.
+/// In tests, you should fill the model with repos, teams, people etc.,
+/// and then call `gh_model` to construct a corresponding GitHubModel.
+/// After that, you can modify the data model further, then generate a diff
+/// and assert that it has the expected value.
+#[derive(Default, Clone)]
+pub struct DataModel {
+    people: Vec<Person>,
+    teams: Vec<TeamData>,
+}
+
+impl DataModel {
+    pub fn user(&mut self, name: &str) -> usize {
```

Comment:
Nit: perhaps we should prefix these mutating methods with something like `add_` so that it's clearer what they're doing.